### ISSUE TITLE
when i use decorate a instance func with my own decorator,it raise W0211

### ISSUE THREAD

**Issue by wqh17101** [PyCQA/pylint#5441]
### Bug description

![image](https://user-images.githubusercontent.com/26429138/143975473-5b1bc52d-596f-4b64-9d66-dae222601a29.png)
it raise W0211: Static method with 'self' as first argument (bad-staticmethod-argument)

### Command used

```shell
pylint a.py --extension-pkg-whitelist=cv2
```

### Pylint output

```shell
W0211: Static method with 'self' as first argument (bad-staticmethod-argument)
```

### Expected behavior

do not  raise this

### Pylint version

```shell
pylint 2.11.1
astroid 2.8.6
Python 3.8.8 (default, Apr 13 2021, 15:08:03) [MSC v.1916 64 bit (AMD64)]
```

**Comment by Pierre-Sassoulas** [comment]
Hello, thank you for opening the issue. Could you provide the code as text instead of as an image ? Text is crawl-able by search engines and readable by visually impaired persons while images are not.

**Comment by wqh17101** [comment]
```
#!/usr/bin/env python
# _*_coding:utf-8_*_
"""
@Time   :  2021/11/30 23:45
@Author :  Qinghua Wang
"""
from functools import wraps, partial


class Dec:
    @staticmethod
    def timing(func=None, *, prefix=""):
        if prefix == "":
            return partial(Dec.timing, prefix="hello")

        @wraps(func)
        def wrapper(*args, **kwargs):
            print(func.__name__)
            res = func(*args, **kwargs)
            return res

        return wrapper


class About:
    def __init__(self):
        self.value = 1

    @Dec.timing()
    def run(self):
        self.func()

    def func(self):
        print(self.value)


if __name__ == '__main__':
    A = About()
    A.run()

```

************* Module pylint_test
pylint_test.py:11:0: C0115: Missing class docstring (missing-class-docstring)
pylint_test.py:13:4: C0116: Missing function or method docstring (missing-function-docstring)
pylint_test.py:11:0: R0903: Too few public methods (1/2) (too-few-public-methods)
pylint_test.py:26:0: C0115: Missing class docstring (missing-class-docstring)
pylint_test.py:31:4: C0116: Missing function or method docstring (missing-function-docstring)
pylint_test.py:31:4: W0211: Static method with 'self' as first argument (bad-staticmethod-argument)
pylint_test.py:34:4: C0116: Missing function or method docstring (missing-function-docstring)
pylint_test.py:40:4: E1120: No value for argument 'self' in staticmethod call (no-value-for-parameter)

------------------------------------------------------------------
it told me that the func `run` had problems w0211,but it is not a staticmethod
